research/conditional/utils/argparse.py

```diff
@@ -4,74 +4,113 @@
 def introduce_parser_arguments(
     parser: argparse.ArgumentParser,
 ) -> argparse.ArgumentParser:
-    # core hyperparameters, fixed for all experiments; needs a good reason to change
+    # CORE model hyperparameters, almost always specified in baseline configs
+    parser.add_argument("--model_type", type=str, choices=["gpt", "bert"])
+    parser.add_argument("--ff_mode", type=str, default="vanilla")
+    parser.add_argument("--n_blocks", type=int, required=True)
+    parser.add_argument("--dmodel", type=int, required=True)
+    parser.add_argument("--dff", type=int, required=True)
+    parser.add_argument("--n_att_heads", type=int, required=True)
+    parser.add_argument("--dhead", type=int, default=None)
```

Comment:
Is default=None sam as required=False

Reply:
no; btw I have removed all required=False